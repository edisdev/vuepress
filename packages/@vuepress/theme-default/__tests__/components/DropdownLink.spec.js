import { mount, RouterLinkStub } from '@vue/test-utils'
import DropdownLink from '../../components/DropdownLink.vue'
import { createLocalVue } from '@vuepress/test-utils/client'

describe('DropdownLink', () => {
  test('renders dropdown link.', () => {
    const item = {
      text: 'VuePress',
      items: [
        {
          text: 'Guide',
          link: '/guide/'
        },
        {
          text: 'Config Reference',
          link: '/config/',
          notFocusable: true
        }
      ]
    }
    const wrapper = mount(DropdownLink, {
      localVue: createLocalVue(),
      stubs: {
        'router-link': RouterLinkStub
      },
      propsData: { item, dropdownName: 'Languages' }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
