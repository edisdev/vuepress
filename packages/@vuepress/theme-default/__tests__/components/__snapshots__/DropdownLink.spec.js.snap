// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownLink renders dropdown link. 1`] = `
<div class="dropdown-wrapper"><button type="button" aria-label="Languages" class="dropdown-title"><span class="title">VuePress</span> <span class="arrow right"></span></button>
  <ul class="nav-dropdown" style="display: none;" name="dropdown">
    <li class="dropdown-item">
      <!----> <a class="nav-link">Guide</a></li>
    <li class="dropdown-item">
      <!----> <a tabindex="-1" class="nav-link">Config Reference</a></li>
  </ul>
</div>
`;
